pulser-core/pulser/sequence/_seq_drawer.py

```diff
@@ -707,11 +726,136 @@ def phase_str(phi: float) -> str:
             if ax_lims[i][0] < 0:
                 ax.axhline(0, **hline_kwargs)
 
+    return (fig_reg if register else None, fig, ch_axes, data)
+
+
+def draw_samples(
+    sampled_seq: SequenceSamples,
+    register: Optional[BaseRegister] = None,
+    sampling_rate: Optional[float] = None,
+    draw_phase_area: bool = False,
+    draw_phase_shifts: bool = False,
+    draw_input: bool = True,
+    draw_modulation: bool = False,
+    draw_phase_curve: bool = False,
+) -> tuple[Figure | None, Figure]:
+    """Draws a SequenceSamples.
+
+    Args:
+        sampled_seq: The input sequence of operations on a device.
+        register: If present, draw the register before the pulse
+            sequence, with a visual indication (square halo) around the qubits
```

Comment:
register: If present, draw the register before the pulse
            sequence samples, with a visual indication (square halo) around the qubits